File: QViz.Java/src/main/java/com/periscope/qviz/json/Tool.java
package com.periscope.qviz.json;

import com.fasterxml.jackson.annotation.JsonRawValue;

/**
 * JSON Model for Tool Integration Details
 */
public class Tool {

	/**
	 * Identifier of the third-party Tool Integration
	 */
	public String toolId;

	/**
	 * Name of the third-party Tool Integration
	 */
	public String toolName;

	/**
	 * Description of the third-party Tool Integration
	 */
	public String description;

	/**
	 * Type of the third-party Tool Integration
	 */
	public String toolType;

	/**
	 * JSON Value of the Configuration Template for the third-party Tool Integration
	 */
	@JsonRawValue
	public String configTemplate;

	/**
	 * Is this third-party Tool Integration Active?
	 */
	public boolean isActive;

	/**
	 * Default Constructor
	 */
	public Tool() {
		this.toolId = "";
		this.toolName = "";
		this.description = "";
		this.toolType = "";
		this.configTemplate = "";
	}

}

File: QViz.NET/Periscope.QViz/JSON/Tool.cs
namespace Periscope.QViz.JSON
{
	/// <summary>
	/// JSON Model for Tool Integration Details
	/// </summary>
	public class Tool
	{
		/// <summary>
		/// Identifier of the third-party Tool Integration
		/// </summary>
		public string toolId;

		/// <summary>
		/// Name of the third-party Tool Integration
		/// </summary>
		public string toolName;

		/// <summary>
		/// Description of the third-party Tool Integration
		/// </summary>
		public string description;

		/// <summary>
		/// Type of the third-party Tool Integration
		/// </summary>
		public string toolType;

		/// <summary>
		/// JSON Value of the Configuration Template for the third-party Tool Integration
		/// </summary>
		public string configTemplate;

		/// <summary>
		/// Is this third-party Tool Integration Active?
		/// </summary>
		public bool isActive;
		/// <summary>
		/// Associated value for tool
		/// </summary>
		public string value { get; set; }

		/// <summary>
		/// Default Constructor
		/// </summary>
		public Tool()
		{
			this.toolId = "";
			this.toolName = "";
			this.description = "";
			this.toolType = "";
			this.configTemplate = "";
			this.value = "";
		}
	}
}